go/libkb/delegatekeyaggregator.go

```diff
@@ -56,3 +66,28 @@ func DelegatorAggregator(lctx LoginContext, ds []Delegator, sdhBoxes []keybase1.
 	_, err = apiArgBase.G().API.PostJSON(apiArg)
 	return err
 }
+
+// Make the "per_user_key" section of an API arg.
+// Modifies `serverArg`.
+func AddPerUserKeyServerArg(serverArg JSONPayload, generation keybase1.PerUserKeyGeneration,
+	pukBoxes []keybase1.PerUserKeyBox, pukPrev *PerUserKeyPrev) {
+	section := make(JSONPayload)
+	if len(pukBoxes) > 0 {
+		section["boxes"] = pukBoxes
+		section["generation"] = pukBoxes[0].Generation
+	}
+	if pukPrev != nil {
+		section["secretbox"] = pukPrev
+	}
+	if len(section) > 0 {
+		serverArg["per_user_key"] = section
+	}
```

Comment:
Just the second one: no prev when adding a new device. The first and third are basically the same and just make it so this method can be called as no-op. But I don't like that anymore, so I'll change it to require boxes.